src/commands/dm.ts

```diff
@@ -0,0 +1,353 @@
+                    await sendDMs(message, users);
+                    return;
+                }
+            })
+        }
+        //if there is a message and no users, or users and no message - specified options.
+        else if (message && users.length < 1 || !message && users.length > 0) {
+            await interaction.reply('If you specify a message, you must also specify at least one user (and vice-versa).');
+            return;
+        }
+        //if no message or user options were specified
+        else {
+            const startEmbed = new EmbedBuilder()
+                .setTitle('DM User(s)')
+                .addFields(
+                    { name: 'Please use the buttons below to: ', value: '- Create your desired message.' },
+                    { name: ' ', value: '- Choose which user(s) you would like the bot to send the message to.' },
+                    { name: ' ', value: '- Finally, click the "Send" button when you are finished.' }
+                );
+
+            const addMsgEmbed = new EmbedBuilder()
+                .setTitle('Create/Change Message')
+                .setDescription('Please type the message you would like me to send for you! Click "Done" when you are satisfied with the message.');
+
+            const addUsersEmbed = new EmbedBuilder()
+                .setTitle('Add Users')
+                .setDescription('Please type the names of the users you\'d like me to DM!');
+
+            const startRow = new ActionRowBuilder<ButtonBuilder>()
+                .setComponents(BUTTONS.CANCEL_BUTTON, BUTTONS.ADD_MESSAGE_BUTTON, BUTTONS.ADD_USERS_BUTTON, BUTTONS.SEND_BUTTON);
+
+            const addMsgRow = new ActionRowBuilder<ButtonBuilder>()
+                .setComponents(BUTTONS.BACK_BUTTON, BUTTONS.DONE_BUTTON);
+
+            const addUsersRow = new ActionRowBuilder<ButtonBuilder>()
+                .setComponents(BUTTONS.BACK_BUTTON, BUTTONS.DONE_BUTTON, BUTTONS.RESET_BUTTON);
+
+            const confirmRow = new ActionRowBuilder<ButtonBuilder>()
+                .setComponents(BUTTONS.BACK_BUTTON, BUTTONS.YES_BUTTON);
+
+
+            const embedMessage = await interaction.reply({ content: 'Please follow the instructions below!', embeds: [startEmbed], components: [startRow] });
+
+            if (!embedMessage) {
+                await interaction.reply('An error has occurred, please contact a developer.');
+                return;
+            }
+
+            const buttonCollector = embedMessage.createMessageComponentCollector({
+                filter: i => i.user.id === interaction.user.id,
+                componentType: ComponentType.Button,
+                time: TIME_IN_MS.TEN_MINUTES
+            });
+
+            let msgCollector: MessageCollector;
+            let userCollector: MessageCollector;
+            let memberList: User[] = [];
+            let DM = '';
+            const msgTooLong = '**__Message too long! (limit: 1024 characters)__**';
```

Comment:
Put this outside of the function (like below the `MAX_USERS` const, with same naming convention, i.e., `MSG_TOO_LONG`)